src/widget/wdoublespinbox.cpp

```diff
@@ -0,0 +1,25 @@
+#include "widget/wdoublespinbox.h"
+
+#include <QLineEdit>
+#include <QLocale>
+
+#include "moc_wdoublespinbox.cpp"
+
+WDoubleSpinBox::WDoubleSpinBox(QWidget* pParent)
+        : QDoubleSpinBox(pParent),
+          m_decSep(QLocale().decimalPoint()) {
+    lineEdit()->setValidator(new QRegExpValidator(
+            QRegExp("[0-9]{1,5}[." +
+                    // add locale decimal point if it's not a dot
+                    (m_decSep != '.' ? m_decSep : QChar()) +
```

Comment:
Could it be worth to simply allow both `,` and `.`? I know my current local would be `.` but I would still use `,` as I grew up using `,`.

I guess the risk is that some users may type `1,000` for `1000`, but in the meantime, how often do you set the BPM value to > 999? :sweat_smile:

Reply:
same as https://github.com/mixxxdj/mixxx/pull/13068#discussion_r1624661452